### PR TITLE
holdings: implement record availability

### PR DESCRIPTION
**Pull request by BadrAly**
* Adds holding availability.
* Provides availability loan conditions for holdings records.

Co-Authored-by: Laurent Dubois <laurent.dubois@uclouvain.be>
Co-Authored-by: Aly Badr <aly.badr@rero.ch>

### CODE REVIEW

File: rero_ils/modules/holdings/api_views.py [hunk 1]
```diff
@@ -20,10 +20,68 @@
```
Comment (rerowep) [inline rero_ils/modules/holdings/api_views.py]: Could we rename this file to `views.py`?

File: rero_ils/modules/holdings/api_views.py [hunk 2]
```diff
@@ -20,10 +20,68 @@
...
+from ...permissions import librarian_permission
 
 api_blueprint = Blueprint(
     'api_holding',
     __name__,
     url_prefix='/holding'
 )
+
+
+def check_authentication(func):
```
Comment (rerowep) [inline rero_ils/modules/holdings/api_views.py]: Do we need this function in other places?

File: rero_ils/modules/holdings/api_views.py [hunk 3]
```diff
@@ -20,10 +20,68 @@
...
+        if not current_user.is_authenticated:
+            return jsonify({'status': 'error: Unauthorized'}), 401
+        if not librarian_permission.require().can():
+            return jsonify({'status': 'error: Forbidden'}), 403
+        return func(*args, **kwargs)
+
+    return decorated_view
+
+
+def jsonify_error(func):
```
Comment (rerowep) [inline rero_ils/modules/holdings/api_views.py]: Do we need this function in other places?

File: rero_ils/modules/holdings/api_views.py [hunk 4]
```diff
@@ -20,10 +20,68 @@
...
+    """Jsonify errors."""
+    @wraps(func)
+    def decorated_view(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except NotFound as e:
```
Comment (rerowep) [inline rero_ils/modules/holdings/api_views.py]: Do we have to use `jsonify` for  `NotFound`?

**Comment by BadrAly** [comment]
> is the term 'interface' in the PR name correct ?

fixed

**Pull request merged**